lms/templates/courseware/course_navigation.html

```diff
@@ -34,21 +39,34 @@
                     <form action="#" class="action-preview-form" method="post">
                         <label for="action-preview-select" class="action-preview-label">${_("View this course as:")}</label>
                         <select class="action-preview-select" id="action-preview-select" name="select">
-                            <option value="staff" ${"selected" if not is_student_masquerade else ""}>${_("Staff")}</option>
-                            <option value="student" ${"selected" if is_student_masquerade and not masquerade_group_id else ""}>${_("Student")}</option>
+                            <option value="staff" ${staff_selected}>${_("Staff")}</option>
+                            <option value="student" ${student_selected}>${_("Student")}</option>
+                            <option value="specific student" ${specific_student_selected}>${_("Specific student")}</option>
                             % if cohorted_user_partition:
                                 % for group in sorted(cohorted_user_partition.groups, key=lambda group: group.name):
-                                <option value="group.id" data-group-id="${group.id}" ${"selected" if masquerade_group_id == group.id else ""}>
+                                <option value="group.id" data-group-id="${group.id}" ${selected(masquerade_group_id == group.id)}>
                                     ${_("Student in {content_group}").format(content_group=group.name)}
                                 </option>
                                 % endfor
                             % endif
                         </select>
-                        <button type="submit" class="sr" name="submit" value="submit">${_("set preview mode")}</button>
+                        <div class="action-preview-username-container">
+                          <label for="action-preview-username" class="action-preview-label">${_("User name or email:")}</label>
+                          <input type="text" class="action-preview-username" id="action-preview-username">
+                        </div>
+                        <button type="submit" class="sr" name="submit" value="submit">${_("Set preview mode")}</button>
                     </form>
                 </li>
             </ol>
         </div>
+        % if specific_student_selected:
+        <div class="preview-specific-student-notice">
+          <p>
+            ${_("You are now viewing the course as <i>{user_name}</i>. Any changes you make will not be saved, "
```

Comment:
@catong I was worried that using the term "read-only view" would suggest that people can't really interact with problems on the page, while in fact they can, but thinking about it, I think the next sentence about changes being reverted on refresh makes it pretty clear that you *can* make temporary changes, so I think your first suggestion is good.

To clarify that the debug tools actually do affect student state, I'd suggest to add a clarifying sentence to the debug tools – I guess this will make it more obvious which statement applies to what.